firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionLifecycleNotifier.java

```diff
@@ -235,22 +235,27 @@ <A extends Activity> Task<Activity> getForegroundActivity(@Nullable Class<A> cla
         return Tasks.forResult(
             getActivityWithIgnoredClass(currentActivity, previousActivity, classToIgnore));
       }
+    }
```

Comment:
`synchronized (lock)` above now ends here, instead of surrounding the entire method.